auth0/src/main/java/com/auth0/android/authentication/storage/CryptoUtil.java

```diff
@@ -85,7 +86,15 @@ public CryptoUtil(@NonNull Context context, @NonNull Storage storage, @NonNull S
             keyStore.load(null);
             if (keyStore.containsAlias(KEY_ALIAS)) {
                 //Return existing key
-                return (KeyStore.PrivateKeyEntry) keyStore.getEntry(KEY_ALIAS, null);
+
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
```

Comment:
move all this block to a new method since you're using the same one a few lines below.
```java
private KeyStore.PrivateKeyEntry getKeyEntryForAlias(KeyStore keyStore, String alias) {
}
```

I suggested that name since the current method this is in is already named similarly.

Reply:
@lbalmaceda I named the method `getKeyEntryCompat` since it is a wrapper that is only checking the API level to call the appropriate methods.

I figured out a method that gets all the tests to run, and gives no warnings on the emulator (I don't have an Android P device, but I believe it should have the same results). However, it is a bit hackish, so I wanted to run it by you to see if it's ok.

I added a condition to check if the `PrivateKey` is `null` at [this line](https://github.com/auth0/Auth0.Android/pull/203/files#diff-84689992633df9060c3e7070601bcf26R156).  Since I couldn't mimic an empty PrivateKey item despite looking into the source codes of some of the classes, I am simply falling back and using `getEntry` as before in such cases.

From what I observed, the `PrivateKey` object is `null` at that point only during the tests. On devices, the `getRSAKeyEntry` method generates the entry if the alias is not present in the keystore.